### PR TITLE
Apply automated sourcery refactoring (`sourcery review . --fix`)

### PR DESCRIPTION
**Pull request by Sigmanificient**
- inline return statements
- use `yield from` pattern
- replaced `.format` with f-string
- use `_` for loop variable when it is unused
- bound `except` to `Exception` instead of anything
- rewrite some block with list/set expression
- simplify some if conditions
- use `contextmanager.suppress` when skipping the exception

### CODE REVIEW

File: docs/screenshots/take_all.py [hunk 1]
```diff
@@ -428,9 +426,7 @@ def take(name, layout, spec):
     except Exception:
         errors.append("While cleaning layout:\n" + traceback.format_exc())
 
-    if errors:
-        return False, "\n\n".join(errors)
-    return True, ""
+    return (False, "\n\n".join(errors)) if errors else (True, "")
```
Comment (jwijenbergh) [inline docs/screenshots/take_all.py:429]: don't like this one

Comment (elParaguayo) [reply docs/screenshots/take_all.py:429]: Yeah. That's nasty.

**Comment by elParaguayo** [comment]
We looked at this before (#4334) and decided against it.

File: libqtile/backend/x11/core.py [hunk 2]
```diff
@@ -333,13 +332,6 @@ def _xpoll(self) -> None:
                         self._motion_notify = None
                     self.handle_event(event)
 
-            # Catch some bad X exceptions. Since X is event based, race
```
Comment (jwijenbergh) [inline libqtile/backend/x11/core.py:336]: it still removes comments :(

Comment (Sigmanificient) [reply libqtile/backend/x11/core.py:336]: Oh that bad!

**Comment by Sigmanificient** [comment]
> I can't see this getting merged.

Hi, this pr is still in work in progress, and have already been quite lighten/improved since this comment.
I hope you'll accept to give it another look when it will be ready

**Comment by tych0** [comment]
The same is true of the last time we tried this: if we're going to merge it, it needs to be enforced in CI, otherwise we'll just diverge again. But I don't love patches like this to begin with since they make traversing git history painful, so I'd like to see some arguments for why listening to sourcery in addition to black is necessary.

**Pull request closed without merging**